### DISCUSSION TITLE
Issue with TBR displaying on the graph

### DISCUSSION THREAD

**Discussion by Artemdrip** [NightscoutFoundation/xDrip#2709] [Q&A]
When using Omnipod Dash in master branch or Accu check Combo with the new combo driver in the APPS dev branch 3.1.0.3, TBR is displayed incorrectly in Xdrip (last nightly version).
AAPS sends TBR with local broadcast to Xdrip and those are displayed with multiple vertical lines.
However if Accu check combo is used with ruffy, everything is displayed correctly.
Unfortunately I don't know if it's due to AAPS or Xdrip.
![a6a375b7-0bb8-46aa-9f91-0bba2d7f1858](https://user-images.githubusercontent.com/69725840/221949275-4d79383a-c1a4-4d56-be9e-69801eecfb08.jpg)
![fb128aaf-0a6d-4346-aeac-2e4438f89ef5](https://user-images.githubusercontent.com/69725840/221949281-3e992b84-390e-4fde-bd50-f0902c2de69d.jpg)
![706c9431-d6b4-412a-9b6f-a89b685638e5](https://user-images.githubusercontent.com/69725840/221949284-5642ac18-c0d8-4298-94d0-98497c08dff5.jpg)
![e058a56d-8cbf-472a-999a-90baa4b0e29b](https://user-images.githubusercontent.com/69725840/221949285-9bcc5765-d509-485a-88b2-17beff732ca9.jpg)

**Comment by Navid200** [comment]
@gruoner Do you know what this could be?

**Reply by gruoner** [reply]
sorry; i'm not familar with calculating and displaying TBR and integrating AAPS.... as far as i can remember the procedure drawing the actual glucose and insulin graphs does not draw a kind of TBR. There must be another function involved - maybe is belonging to the history.
@Artemdrip does this failure also occur on the main screen showing the current values?

**Reply by Artemdrip** [reply]
@gruoner The issue is also in the main screen. You can see it in the third screenshot.
I think @tzachi-dar had a PR for TBR  #1303 and #1529.  I don't know if the current TBR function was integrated by him or existed before.
